geoapps/grid_creation/driver.py

```diff
@@ -0,0 +1,85 @@
+#  Copyright (c) 2022 Mira Geoscience Ltd.
+#
+#  This file is part of geoapps.
+#
+#  geoapps is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+from __future__ import annotations
+
+import os
+import sys
+from os import path
+
+import numpy as np
+from geoh5py.ui_json import InputFile, monitored_directory_copy
+from scipy.spatial import cKDTree
+
+from geoapps.grid_creation.params import GridCreationParams
+from geoapps.interpolation.application import DataInterpolation
+from geoapps.shared_utils.utils import get_locations
+
+
+class GridCreationDriver:
+    def __init__(self, params: GridCreationParams):
+        self.params: GridCreationParams = params
+
+    def run(self):
+        xyz = get_locations(self.params.geoh5, self.params.objects)
+        if xyz is None:
+            raise ValueError("Input object has no centroids or vertices.")
+
+        tree = cKDTree(xyz)
+
+        xyz_ref = get_locations(self.params.geoh5, self.params.xy_reference)
+        if xyz_ref is None:
+            print("No object selected for 'Lateral Extent'. Defaults to input object.")
+            xyz_ref = xyz
+
+        # Find extent of grid
+        h = [self.params.cell_size_x, self.params.cell_size_y, self.params.cell_size_z]
+        # pads: W, E, S, N, D, U
+        pads = [
+            self.params.horizontal_padding,
+            self.params.horizontal_padding,
+            self.params.horizontal_padding,
+            self.params.horizontal_padding,
+            self.params.bottom_padding,
+            0.0,
+        ]
+
+        object_out = DataInterpolation.get_block_model(
```

Comment:
Seems like the get_block_model method should be stored in the block model application and called by the interpolation app, not the other way around.